samples/azure/azure-service-fabric-routing/Core_6/ZipCodeVoteCount/ZipCodeVotes.cs

```diff
@@ -0,0 +1,57 @@
+using System;
+using System.Threading.Tasks;
+using NServiceBus;
+
+public class ZipCodeVotes :
+    Saga<ZipCodeVotes.ZipCodeVoteData>,
+    IAmStartedByMessages<TrackZipCode>,
+    IHandleTimeouts<CloseVoting>
+{
+    protected override void ConfigureHowToFindSaga(SagaPropertyMapper<ZipCodeVoteData> mapper)
+    {
+        mapper.ConfigureMapping<TrackZipCode>(m => m.ZipCode)
+            .ToSaga(s => s.ZipCode);
+    }
+
+    public async Task Handle(TrackZipCode message, IMessageHandlerContext context)
+    {
+        if (!Data.Started)
+        {
+            await RequestTimeout<CloseVoting>(context, DateTime.UtcNow.AddMinutes(1))
+                .ConfigureAwait(false);
+            Data.ZipCode = message.ZipCode;
+            Data.Started = true;
+        }
+
+        Data.Count++;
```

Comment:
Shouldn't the voting endtime be stored in the saga instance and checked before increasing? Currently it is assumed the timeout will be triggered immediately.